window/src/os/windows/window.rs

```diff
@@ -716,72 +735,129 @@ impl WindowOps for Window {
-                lfUnderline: 0,
-                lfStrikeOut: 0,
-                lfCharSet: 0,
-                lfOutPrecision: 0,
-                lfClipPrecision: 0,
-                lfQuality: 0,
-                lfPitchAndFamily: 0,
-                lfFaceName: [0u16; 32],
-            };
-            let theme = OpenThemeData(
-                hwnd,
-                [
-                    'H' as u16, 'E' as u16, 'A' as u16, 'D' as u16, 'E' as u16, 'R' as u16, 0u16,
-                ]
-                .as_ptr(),
-            );
-            if !theme.is_null() {
-                let res = GetThemeFont(
-                    theme,
-                    hdc,
-                    HP_HEADERITEM,
-                    HIS_NORMAL,
-                    TMT_CAPTIONFONT,
-                    &mut log_font,
-                );
-                if res == S_OK {
-                    CloseThemeData(theme);
-                    return Some(log_font);
-                }
-            }
+    fn get_os_parameters(&self, config: &ConfigHandle) -> anyhow::Result<Option<Parameters>> {
+        let hwnd = self.0 .0;
+        if hwnd.is_null() {
+            return Err(anyhow::anyhow!("HWND is null"));
+        }
 
-            let res = GetThemeSysFont(theme, TMT_CAPTIONFONT, &mut log_font);
-            if !theme.is_null() {
-                CloseThemeData(theme);
-            }
+        let has_focus = !unsafe { GetFocus() }.is_null();
+        let is_maximized = window_is_maximized(hwnd);
 
-            if res == S_OK {
-                Some(log_font)
+        let is_full_screen = unsafe {
+            let mut mi = MONITORINFO {
+                cbSize: std::mem::size_of::<MONITORINFO>() as u32,
+                ..Default::default()
+            };
+            if GetMonitorInfoW(MonitorFromWindow(hwnd, MONITOR_DEFAULTTOPRIMARY), &mut mi)
+                == winapi::shared::minwindef::TRUE
+            {
+                let mut client_rect = RECT::default();
+                if GetClientRect(hwnd, &mut client_rect) == winapi::shared::minwindef::TRUE {
+                    client_rect.right >= mi.rcMonitor.right - mi.rcMonitor.left
+                        && client_rect.bottom >= mi.rcMonitor.bottom - mi.rcMonitor.top
+                } else {
+                    false
+                }
             } else {
-                None
+                false
             }
-        }
-        unsafe {
-            let hwnd = self.0 .0;
+        };
+
+        let title_font = unsafe {
             let hdc = GetDC(hwnd);
-            let result = match populate_log_font(hwnd, hdc) {
-                Some(lf) => wezterm_font::locator::gdi::parse_log_font(&lf, hdc).ok(),
+            if hdc.is_null() {
+                return Err(anyhow::anyhow!("Could not get device context"));
+            }
+            let result = match get_title_log_font(hwnd, hdc) {
+                Some(lf) => Some(wezterm_font::locator::gdi::parse_log_font(&lf, hdc)?),
```

Comment:
The `?` can cause us to return from here without releasing `hdc` and leak it.

I'd suggest going back to the prior implementation of this bit; I don't think there's a ton of value in propagating an error here, as the code already handles the `None` case.